nibabel/loadsave.py

```diff
@@ -40,55 +33,15 @@ def load(filename, **kwargs):
     img : ``SpatialImage``
        Image of guessed type
     '''
-    return guessed_image_type(filename).from_filename(filename, **kwargs)
 
+    sniff = None
+    for image_klass in all_image_classes:
+        is_valid, sniff = image_klass.is_image(filename, sniff)
+        if is_valid:
+            return image_klass.from_filename(filename, **kwargs)
 
-def guessed_image_type(filename):
```

Comment:
Both, I think.  Or do you think someone would want this function independent of the `load` mechanism?